### DISCUSSION TITLE
json document scheme

### DISCUSSION THREAD

**Discussion by username872513** [Koenkk/zigbee2mqtt#10085] [Devices]
Hello everyone,
I am currently working on my thesis where Zigbee2MQTT takes a big part. I would like to know if there is a json document with all the devices and their semantic or the json scheme, so it can be processed automatically.

I am a big fan of Zigbee2MQTT and I really appreciate the work!

I hope you can help me.
Best regards

**Comment by Koenkk** [comment]
Could you clarify better what you are looking for?

**Reply by username872513** [reply]
The task of a my student research project is the adaptation of the WoT model to the Zigbee2MQTT gateway. After doing some first practical research with the Zigbee2MQTT project, I think that this is a very exciting project.

For a generic implementation it would be very helpful for us if we had a JSON (schema) description of all exposes available (like this: https://www.zigbee2mqtt.io/guide/usage/exposes.html#access - only for all exposes and features). 

We have already created a JSON document** of all supported devices from the following script (source: https://github.com/Koenkk/zigbee2mqtt.io/tree/master/docs/supported-devices). This includes general device descriptions and a list of exposes, which helps us a lot. However, for more complex exposes, e.g. "light", the references to the "features" that a device actually supports are missing - this information or rferences would also be very helpful for us.

Maybe there is a possibility to provide us with this information or is there a possibility that we could simply extract this information ourselves.

I hope that these further explanations will be helpful for a better understanding regarding what we want to develop. If you have any further questions, please feel free to contact us.  

Thank you very much for your support, best regards,

**
...
{
    "model": "E1F-N9G",
    "vendor": "Sengled",
    "addedAt": "2021-11-30T20:10:17",
    "description": "Smart LED filament candle (E12)",
    "image": "../images/devices/E1F-N9G.jpg",
    "link": "../devices/E1F-N9G.html",
    "exposes": [
      "light"
    ]
  },

**Reply by Koenkk** [reply]
Such information is not available but you should be able to extract it yourself, example of a test which accesses such info: https://github.com/Koenkk/zigbee-herdsman-converters/blob/10f29e0be3aad30807c0e469e4095ef5c58b4d0c/test/index.test.js#L406
